### PR TITLE
Use system libraries if available

### PR DESCRIPTION
**Pull request by superm1**
Rather than relying upon pulling libraries from the web, many distros have them already available in the repos.  Use those if possible.

### CODE REVIEW

**Comment by superm1** [comment]
NOTE: I expect that this won't work until https://github.com/lemonade-sdk/lemonade/pull/911 is merged.

**Comment by jeremyfowers** [comment]
> NOTE: I expect that this won't work until #911 is merged.

merged! and updated here.

File: src/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -28,45 +28,55 @@ endif()
...
+set(USE_SYSTEM_CURL ${CURL_FOUND})
+set(USE_SYSTEM_ZSTD ${ZSTD_FOUND})
+set(USE_SYSTEM_CLI11 ${CLI11_INCLUDE_DIRS})
+set(USE_SYSTEM_HTTPLIB ${HTTPLIB_FOUND})
+
+if(USE_SYSTEM_JSON)
+    message(STATUS "Using system nlohmann_json")
+else()
+    message(STATUS "System nlohmann_json not found, will use FetchContent")
+endif()
```
Comment (jeremyfowers) [inline src/cpp/CMakeLists.txt:93]: There's no notion of required version numbers for the system packages (unless I missed something).

What happens here there's a very old version of the dependence installed on the system?

Comment (superm1) [reply src/cpp/CMakeLists.txt:93]: You're right.  This is missing.  Let me see if I can come up with a way to detect the versions from pkg config and only allow system ones if new enough.

FWIW I was able to build on both Debian unstable and Ubuntu resolute as long as I pulled an updated cpp-httplib, but on Ubuntu noble or Debian stable I expect failures, so this will be a good unit test to confirm it works.

**Comment by superm1** [comment]
> Needs to handle the corner case: What happens here there's a very old version of the dependence installed on the system?

Should be sorted now. Introspecting all dep versions.

File: src/cpp/CMakeLists.txt [hunk 2]
```diff
@@ -77,63 +133,93 @@ set(BUILD_SHARED_LIBS OFF CACHE INTERNAL "")  # Build static libraries
...
+
 # === nlohmann/json ===
-set(JSON_BuildTests OFF CACHE INTERNAL "")
-set(JSON_Install OFF CACHE INTERNAL "")
-FetchContent_MakeAvailable(json)
+if(NOT USE_SYSTEM_JSON)
+    FetchContent_Declare(json
+        GIT_REPOSITORY https://github.com/nlohmann/json.git
+        GIT_TAG v3.11.3
```
Comment (jeremyfowers) [inline src/cpp/CMakeLists.txt]: Can you use the constant from line 35 instead of having two sources for this number?

Comment applies to all 5 packages.
